### PR TITLE
Main

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

File: src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        <property name="connection.password">Oleg066R</property>
+        <property name="current_session_context_class">thread</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+        <mapping class="mate.academy.model.CinemaHall"/>
+        <mapping class="mate.academy.model.Movie"/>
+        <mapping class="mate.academy.model.MovieSession"/>
+        <mapping class="mate.academy.model.User"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (okuzan) [inline src/main/resources/hibernate.cfg.xml:16]: add newline

File: src/main/java/mate/academy/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package mate.academy.util;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
```
Comment (okuzan) [inline src/main/java/mate/academy/util/HashUtil.java:7]: add private constructor

File: src/main/java/mate/academy/util/HashUtil.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            StringBuilder passwordBuilder = new StringBuilder();
+            for (int i = 0; i < digest.length; i++) {
+                passwordBuilder.append(Integer.toHexString(0xFF & digest[i]));
+            }
+            return passwordBuilder.toString();
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Can't get hash password ", e);
+        }
+    }
+
```
Comment (okuzan) [inline src/main/java/mate/academy/util/HashUtil.java:31]:
```suggestion
```

File: src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) {
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:19]: image

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request closed without merging**